unit/outcome_table_decider.py

```diff
@@ -0,0 +1,251 @@
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+
+import unittest
+import unittest.mock
+
+import lib.job_outcome
+
+
+class CBMCNegativeTest(unittest.TestCase):
+    def setUp(self):
+        self.table = {
+            "comment": "",
+            "outcomes": [{
+                "type": "return-code",
+                "value": 0,
+                "action": "fail_ignored",
+            }, {
+                "type": "return-code",
+                "value": 10,
+                "action": "success",
+            }, {
+                "type": "wildcard",
+                "action": "fail",
+            }]
+        }
+
+
+    def test_zero(self):
+        proc = unittest.mock.Mock()
+        proc.returncode = 0
+        jod = lib.job_outcome.OutcomeTableDecider(self.table, proc, False, True)
+
+        self.assertEqual(
+            jod.get_job_fields(), {
+                "outcome": "fail_ignored",
+                "loaded_outcome_dict": self.table,
+            })
+
+
+    def test_success(self):
+        proc = unittest.mock.Mock()
+        proc.returncode = 10
+        jod = lib.job_outcome.OutcomeTableDecider(self.table, proc, False, True)
+
+        self.assertEqual(
+            jod.get_job_fields(), {
+                "outcome": "success",
+                "loaded_outcome_dict": self.table,
+            })
+
+
+
+class CBMCTest(unittest.TestCase):
+    def setUp(self):
+        self.table = {
+            "comment": "",
+            "outcomes": [{
+                "type": "return-code",
+                "value": 0,
+                "action": "success",
+            }, {
+                "type": "return-code",
+                "value": 10,
+                "action": "fail_ignored",
+            }, {
+                "type": "wildcard",
+                "action": "fail",
+            }]
+        }
+
+
+    def test_success(self):
+        proc = unittest.mock.Mock()
+        proc.returncode = 0
+        jod = lib.job_outcome.OutcomeTableDecider(self.table, proc, False, True)
+
+        self.assertEqual(
+            jod.get_job_fields(), {
+                "outcome": "success",
+                "loaded_outcome_dict": self.table,
+            })
+
+
+    def test_ten(self):
+        proc = unittest.mock.Mock()
+        proc.returncode = 10
+        jod = lib.job_outcome.OutcomeTableDecider(self.table, proc, False, True)
+
+        self.assertEqual(
+            jod.get_job_fields(), {
+                "outcome": "fail_ignored",
+                "loaded_outcome_dict": self.table,
+            })
+
+
+    def test_one(self):
+        proc = unittest.mock.Mock()
+        proc.returncode = 1
+        jod = lib.job_outcome.OutcomeTableDecider(self.table, proc, False, True)
+
+        self.assertEqual(
+            jod.get_job_fields(), {
+                "outcome": "fail",
+                "loaded_outcome_dict": self.table,
+            })
```

Comment:
I think we could benefit from having an `assertOutcomeEquals(jod, outcome)` method that makes this call instead. You can then reuse it in many places throughout this code.